runtime/src/main/java/org/corfudb/protocols/CorfuProtocolCommon.java

```
@@ -0,0 +1,209 @@
+    }
+
+    /**
+     * Returns a Layout object from its Protobuf representation.
+     *
+     * @param msg   the Protobuf Layout message
+     */
+    public static Layout getLayout(LayoutMsg msg) {
+        return Layout.fromJSONString(msg.getLayoutJson());
+    }
+
+    /**
+     * Returns the Protobuf representation of a Token, given
+     * its epoch and sequence.
+     *
+     * @param epoch      the epoch of the token
+     * @param sequence   the sequencer number of the token
+     */
+    public static TokenMsg getTokenMsg(long epoch, long sequence) {
+        return TokenMsg.newBuilder()
+                .setEpoch(epoch)
+                .setSequence(sequence)
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a Token.
+     *
+     * @param token   the desired (Java) Token
+     */
+    public static TokenMsg getTokenMsg(Token token) {
+        return getTokenMsg(token.getEpoch(), token.getSequence());
+    }
+
+    /**
+     * Returns the Protobuf representation of a SequencerMetrics object.
+     *
+     * @param metrics   the desired (Java) SequencerMetrics
+     */
+    public static SequencerMetricsMsg getSequencerMetricsMsg(SequencerMetrics metrics) {
+        return SequencerMetricsMsg.newBuilder()
+                .setSequencerStatus(sequencerStatusTypeMap.get(metrics.getSequencerStatus()))
+                .build();
+    }
+
+    /**
+     * Returns a SequencerMetrics object from its Protobuf representation.
+     *
+     * @param msg   the Protobuf SequencerMetrics message
+     */
+    public static SequencerMetrics getSequencerMetrics(SequencerMetricsMsg msg) {
+        switch(msg.getSequencerStatus()) {
+            case READY: return SequencerMetrics.READY;
+            case NOT_READY: return SequencerMetrics.NOT_READY;
+            case UNKNOWN: return SequencerMetrics.UNKNOWN;
+            default:
+        }
+
+        //TODO: Revisit for rolling-upgrades?
+        return null;
```

You might want to throw an `InvalidStateException` insated in the default block.